Add optional timezone parameter to DateTimeProcessor issue/#23

smnv-as opened the pull request:
Яков, я постарался сделать. Проврь, верно ли я понял задачу.
Changes to be committed:
	modified:   import_me/processors.py
	modified:   tests/conftest.py
	modified:   tests/test_processors.py

smnv-as committed:
Add optional timezone parameter to DateTimeProcessor issue/#23

Changes to be committed:
	modified:   import_me/processors.py
	modified:   tests/conftest.py
	modified:   tests/test_processors.py

yakovistomin commented on import_me/processors.py:
```diff
@@ -178,6 +181,12 @@ def process_value(self, value: Any) -> Any:
             raise ColumnError(f'Unable to convert to date {value}.')
+        if self.user_timezone:
+            try:
+                user_timezone = timezone(self.user_timezone)
```
Сделай пожалуйста это в init. Таймзона не меняется для инстанса, нет смысла её для каждого value инициализировать.

smnv-as replied:
@Valerich Да, я хотел изначально так сделать, но у меня не получилось обработку ошибок нормально сделать. Не перехватываются ошибки. Пошел разбираться.

yakovistomin commented on tests/test_processors.py:
```diff
@@ -144,36 +144,44 @@ def test_multiple_processor_none_if_error():
+        (None, None, '20.07.2019 12:43:52', datetime_for_test),
+        (None, [], '20.07.2019 12:43:52', datetime_for_test),
+        (None, '', '20.07.2019 12:43:52', datetime_for_test),
+        ('Europe/Moscow', '', '20.07.2019 12:43:52', datetime_for_test_with_user_timezone),
```
Можно еще тест на то, что будет, если на вход подадут дату уже с таймзоной

yakovistomin commented on import_me/processors.py:
```diff
@@ -165,9 +167,10 @@ def process_value(self, value: Any) -> Any:
 
 class DateTimeProcessor(BaseProcessor):
-    def __init__(self, formats: Collection[str] = None, **kwargs: Any) -> None:
+    def __init__(self, formats: Collection[str] = None, user_timezone: str = None, **kwargs: Any) -> None:
```
давай обзовем параметр 'timezone' и он будет Union[str, timezone]

smnv-as replied:
@Valerich  я правильно понял, надо  чтобы было вот так:  def __init__(self, formats: Collection[str] = None, timezone: Union[str, timezone] = None, **kwargs: Any) -> None: ?

yakovistomin replied:
да, так и есть

Melevir commented:
@Valerich check this out plz

smnv-as committed:
Refactoring issue/#23

 Changes to be committed:
	modified:   .gitignore
	modified:   import_me/processors.py
	modified:   tests/test_processors.py

Melevir commented:
Гайз, этот пр висит уже месяц. Давайте смерджим его, если это ок, или закроем.
@Valerich что скажешь? Можно мерджить или ещё нет?

Melevir commented:
Закрываю.
@dumbturtle спасибо за труд всё равно. Если решишь доделать – сделай новый пр, пожалуйста.
